src/helpers/randomizeCard/index.js

```diff
@@ -0,0 +1,736 @@
+		}
+	}
+
+	fillSlots() {
+		while(true) {
+			var foundState = 0
+			var target = ""
+			var char = ""
+			for (let i=0; i<this.effect.length; i++) {
+				char = this.effect[i]
+				if (char == '{') {
+					foundState = 1
+				}
+				if (foundState == 1 && char != '{' && char != '}') {
+					target += char
+				}
+				if (char == '}' && foundState == 1) {
+					foundState = 2
+					break
+				}
+			}
+			
+			if (foundState != 2) {return}
+
+			var valid = []
+			slots.forEach((slot) => {
+				if (slot.type == target) {valid.push(slot)}
+			})
+				
+			if (this.faction == "Shadowfen") {
+				slotsShadowfen.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Ironclad") {
+				slotsIronclad.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Winter") {
+				slotsWinter.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Swarm") {
+				slotsSwarm.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			
+			var targetSlot = rchoice(valid)
+			
+			this.effect = this.effect.replace("{"+target+"}",targetSlot.part1)
+			this.effect = this.effect.replace("{"+target+"2}",targetSlot.part2)
+			this.effCost *= targetSlot.cost
+			if (verbose) {print(`Adding {target} multiplier of {targetSlot.cost}. New cost: {this.effCost}`)}
+		}
+	}
+
+	getName() {
+		names.forEach((nameList) => {
+			if (nameList[0] == this.tribe.name) {
+				this.firstName = rchoice(nameList[1])
+				this.lastName = rchoice(nameList[2])
+			}
+		})
+	}
+	
+	getImageID() {
+		var imageKey = ["construct","frostling","knight","pirate","raven","rodent","satyr","toad","undead","dwarf","dragon","feline"]
+		for(var i=0; i<imageKey.length; i++) {
+			if (imageKey[i] == this.tribe.name) {
+				return "T"+(i+1).toString()
+			}
+		}
+		return "T3"
+	}
+	
+	printCard() {
+		this.effect = this.effect.replace("	"," ")
+		console.log(`${this.firstName} ${this.lastName}`)
+		console.log(`${this.mana+Math.floor(this.effCost)} mana - ${this.tribe.name} ${this.subtribe}`)
+		console.log(`|	${this.str} / ${this.mov}`)
+		console.log("| " + this.effect)
+		console.log(`${this.faction} - ${this.rarity}`)
+		if (verbose) {console.log(`Actual cost: ${this.mana}+${this.effCost}`)}
+	}
+}
+
+const randomizeCard = () => {
+	var card = new Card()
+	card.generate()
+	var ancientValue = false
+	var elderValue = false
+	if (card.subtribe == "ancient") {ancientValue = true}
+	if (card.subtribe == "elder") {elderValue = true}
+	var imageID = card.getImageID()
+	return {
+		type: 'unit',
+		race: card.tribe.name,
+		rarity: card.rarity.toLowerCase(),
+		faction: card.faction.toLowerCase(),
+	name: card.firstName + "" + card.lastName,
+	//imageCardId: "Token " + card.tribe.name[0].toUpperCase() + card.tribe.name.substring(1),
+	mana: (card.mana + Math.floor(card.effCost)).toString(),
+	strength: card.str.toString(),
```

Comment:
Do you handle leveling yet? As in 1/2/3/4/5?

Reply:
I currently do not handle leveling. I could potentially change the base statlines to handle strength leveling, but I don't see any way to handle effect leveling easily, and that means I likely will be unable to apply leveling effects to the spell cards when I get around to generating them. Should I include leveling, or should we leave that out of the equation?